client/block-builder/src/lib.rs

```diff
@@ -40,9 +45,42 @@ use sp_runtime::{
 	Digest,
 };
 
+use sc_client_api::backend;
+use sc_executor::RuntimeVersionOf;
 pub use sp_block_builder::BlockBuilder as BlockBuilderApi;
+use sp_core::storage::{well_known_keys, StateVersion, Storage};
 
-use sc_client_api::backend;
+/// Return the state version given the genesis storage and executor.
+pub fn resolve_state_version_from_wasm<E>(
```

Comment:
The function is duplicated in [`client/service/src/client/client.rs:1166`](https://github.com/paritytech/substrate/blob/e06586108544f39cb71d365552b80b04daa17fd9/client/service/src/client/client.rs#L1166)

Should we keep it only in one place?

Reply:
Good point and yes!

Reply:
Yeah, I just copy this function. Forget to delete this one.